src/SWP/Bundle/ContentBundle/EventListener/ArticleMetadataSubscriber.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\EventListener;
+
+use SWP\Bundle\ContentBundle\ArticleEvents;
+use SWP\Bundle\ContentBundle\Event\ArticleEvent;
+use SWP\Component\Bridge\Model\PackageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ArticleMetadataSubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ArticleEvents::PRE_CREATE => 'populateMetadata',
+        ];
+    }
+
+    /**
+     * @param ArticleEvent $event
+     */
+    public function populateMetadata(ArticleEvent $event)
+    {
+        $package = $this->getPackage($event);
+        $article = $event->getArticle();
+
+        $article->setMetadata([
```

Comment:
Honestly, I have no idea why you're persisting the Package. Do you use it anywhere? When we ingest the Package, we persist all the data we need from it elsewhere. For me, alarm bells ring when I see the same data being stored twice. If it's an optimization, then do it when you know that it's required. A simple select query is hardly going to make much difference. And normalization is a fundamental principle of database design for a good reason, I guess.